File: backend/src/main/java/com/yourorg/appname/service/RoomService.java
package com.yourorg.appname.service;

import com.yourorg.appname.dto.request.RoomRequest;
import com.yourorg.appname.dto.response.RoomResponse;
import java.util.List;

public interface RoomService {
    List<RoomResponse> getAllRooms(String wing, Integer floor, String status);
    RoomResponse getRoomById(Long id);
    RoomResponse createRoom(RoomRequest request);
    RoomResponse updateRoom(Long id, RoomRequest request);
    RoomResponse assignStudentToRoom(Long roomId, Long studentId, String bedNumber);
    RoomResponse removeStudentFromRoom(Long roomId, Long studentId);
    void deleteRoom(Long id);
}

File: frontend/src/services/roomService.js
import apiClient from './apiClient';

export const roomService = {
  getAllRooms: async (params = {}) => {
    const response = await apiClient.get('/api/rooms', { params });
    return response.data;
  },
  getRoomById: async (id) => {
    const response = await apiClient.get(`/api/rooms/${id}`);
    return response.data;
  },
  createRoom: async (data) => {
    const response = await apiClient.post('/api/rooms', data);
    return response.data;
  },
  updateRoom: async (id, data) => {
    const response = await apiClient.put(`/api/rooms/${id}`, data);
    return response.data;
  },
  assignStudent: async (roomId, studentId, bedNumber) => {
    const response = await apiClient.post(
      `/api/rooms/${roomId}/assign/${studentId}`,
      null,
      { params: { bedNumber } }
    );
    return response.data;
  },
  removeStudent: async (roomId, studentId) => {
    const response = await apiClient.delete(`/api/rooms/${roomId}/remove/${studentId}`);
    return response.data;
  },
};
